eel/tests/sending_payments_test.rs

```diff
@@ -105,12 +113,51 @@ mod sending_payments_test {
             nigiri::query_node_balance(LspdLnd).unwrap()
         );
 
-        // todo also test multipath payments
-    }
+        // test sending mpp
+        let channels_info = node.get_node_info().channels_info;
+        assert_eq!(channels_info.num_usable_channels, 1);
+        assert_eq!(channels_info.local_balance_msat, FOURTY_K_SATS);
+        assert_eq!(channels_info.total_channel_capacities_msat, ONE_M_SATS);
 
-    const THOUSAND_SATS: u64 = 1_000_000;
-    const SECONDS_IN_AN_HOUR: u64 = 3600;
-    const DESCRIPTION_SAMPLE: &str = "Luke, I Am Your Father";
+        let invoice_details = node
+            .create_invoice(NINE_HUNDRED_K_SATS, "test".to_string(), String::new()) // 800 was okay
+            .unwrap();
+        assert!(invoice_details.invoice.starts_with("lnbc"));
+
+        nigiri::pay_invoice(LspdLnd, &invoice_details.invoice).unwrap();
+        assert_eq!(
+            node.get_node_info().channels_info.local_balance_msat,
+            NINE_HUNDRED_K_SATS + FOURTY_K_SATS
+        );
+
+        nigiri::initiate_channel_from_remote(node.get_node_info().node_pubkey, LspdLnd);
+
+        wait_for_eq!(node.get_node_info().channels_info.num_channels, 2);
+        wait_for_eq!(
+            node.get_node_info()
+                .channels_info
+                .total_channel_capacities_msat,
+            ONE_M_SATS * 2
+        );
+
+        let invoice_details = node
+            .create_invoice(NINE_HUNDRED_K_SATS, "test".to_string(), String::new()) // 800 was okay
+            .unwrap();
+        assert!(invoice_details.invoice.starts_with("lnbc"));
+
+        wait_for!(nigiri::pay_invoice(LspdLnd, &invoice_details.invoice).is_ok());
+        assert_eq!(
+            node.get_node_info().channels_info.local_balance_msat,
+            NINE_HUNDRED_K_SATS + NINE_HUNDRED_K_SATS + FOURTY_K_SATS
+        );
+
+        // Node has 2 channels of 2M SAT each. Paying 1.5M SAT requires sending through both of them
+        let invoice = nigiri::issue_invoice(LspdLnd, "MPP", MORE_THAN_ONE_M_SATS, 3600).unwrap();
+
+        node.pay_invoice(invoice, String::new()).unwrap();
+
+        wait_for!(nigiri::query_node_balance(LspdLnd).unwrap() > MORE_THAN_ONE_M_SATS);
```

Comment:
A suggested on the receiving test, we can check for exact values. Also, does it need to be a wait instead of assert?
```suggestion
        wait_for_eq!(nigiri::query_node_balance(LspdLnd).unwrap(), initial_balance + MORE_THAN_ONE_M_SATS);
```

Reply:
Check out e6fe545c4a805c42b5b94372f385beac37611cd6

Reply:
Check out e1b5bb95f72c44629f9ec2a1aa87b128692e3a7c

Yes, the wait is required because the payment happens in another thread, so the assert would be called before the payment actually went through.

Reply:
Ah yes, you're right! Thanks for elaborating